File: app/controllers/facility/RoomLike.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
// SPDX-FileCopyrightText: 2024. The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package controllers.facility;

import play.i18n.Lang;

public interface RoomLike {
    String getRoomInstruction();

    String getRoomInstructionEN();

    String getRoomInstructionSV();

    default String getRoomInstructions(Lang lang) {
        return switch (lang.code()) {
            case "sv" -> {
                String instructions = getRoomInstructionSV();
                yield instructions == null ? getRoomInstruction() : instructions;
            }
            case "en" -> {
                String instructions = getRoomInstructionEN();
                yield instructions == null ? getRoomInstruction() : instructions;
            }
            default -> getRoomInstruction();
        };
    }
}

File: app/models/facility/RoomLike.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.facility

import play.i18n.Lang

trait RoomLike:
  def roomInstruction: String
  def roomInstructionEN: String
  def roomInstructionSV: String

  def getRoomInstructions(lang: Lang): String =
    lang.code() match
      case "sv" => Option(roomInstructionSV).getOrElse(roomInstruction)
      case "en" => Option(roomInstructionEN).getOrElse(roomInstruction)
      case _    => roomInstruction
